### PR TITLE
F3792625/issue76

### PR DESCRIPTION
**Pull request by mkrakowitzer**

### CODE REVIEW

File: examples/jira_mysql_nativessl_install.pp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  } ->
+
+  mysql::db { 'jira':
+    user     => 'jiraadm',
+    password => 'mypassword',
+    host     => 'localhost',
+    grant    => ['ALL'],
+  } ->
+
+  exec { 'tmpkey':
```
Comment (nibalizer) [inline examples/jira_mysql_nativessl_install.pp]: We can't use a snakeoil cert from the OS?

Comment (mkrakowitzer) [inline examples/jira_mysql_nativessl_install.pp]: Does it exist on CentOS and RHEL? I thought it was just in Debian based releases.

Possibly I should remove this exec completely from the example?

Comment (igalic) [inline examples/jira_mysql_nativessl_install.pp]: yeah, just make it jira.example.com.{crt,key}.pem in some phony location.

**Comment by mkrakowitzer** [comment]
@igalic @nibalizer Could I get your opinion on this as I would like to merge it?

Ill rebase it assuming it looks OK.

**Comment by nibalizer** [comment]
I'm hoping to get a big blob of reviews done this weekend and early next week. Can't focus on it at the moment though.

**Comment by mkrakowitzer** [comment]
@ffrank if you could take a look please.

File: README.md [hunk 2]
```diff
@@ -352,6 +350,30 @@ Defaults to '150'
 
 Defaults to '100'
 
+#####`$tomcatNativeSsl`
+
+Enable https/ssl support. Defaults to 'false'. See https://confluence.atlassian.com/display/JIRA/Running+JIRA+over+SSL+or+HTTPS for additional info. The java keystore can be managed with the puppetlabs-java\_ks module or manually with `keytool -genkey -alias jira -keyalg RSA -sigalg SHA256withRSA -keystore /home/jira/jira.ks`
+
+#####`$tomcatHttpsPort`
+
+https/ssl Port to listen on, defaults to 8443, Defaults to ''
```
Comment (ffrank) [inline README.md]: This is ever so slightly confusing.

File: README.md [hunk 3]
```diff
@@ -352,6 +350,30 @@ Defaults to '150'
...
+
+Location of java keystore file. Defaults to '/home/jira/jira.jks'
+
+#####`$tomcatKeystorePass`
+
+Password to access java keystore. Defaults to 'changeit'
+
+#####`$tomcatKeystoreType`
+
+Defaults to 'JKS'.
```
Comment (ffrank) [inline README.md]: What options are available here?

File: manifests/init.pp [hunk 4]
```diff
@@ -107,8 +106,14 @@
...
-  $tomcatAddress = undef,
-  $tomcatPort = 8080,
+  $tomcatAddress      = undef,
+  $tomcatPort         = 8080,
+  $tomcatNativeSsl    = false,
+  $tomcatHttpsPort    = 8443,
+  $tomcatKeyAlias     = 'jira',
+  $tomcatKeystoreFile = '/home/jira/jira.jks',
+  $tomcatKeystorePass = 'changeit',
+  $tomcatKeystoreType = 'JKS',
```
Comment (ffrank) [inline manifests/init.pp]: Not introduced by this PR, but the interface mixes CamelCase, allowercase and snake_case. I feel that we should move modules towards consistently using underscores in parameter names. (Needs not happen in this PR.)

File: manifests/init.pp [hunk 5]
```diff
@@ -132,6 +137,8 @@
   validate_re($contextpath, ['^$', '^/.*'])
   validate_hash($resources)
   validate_hash($ajp)
+  validate_bool($tomcatNativeSsl)
+  validate_absolute_path($tomcatKeystoreFile)
```
Comment (ffrank) [inline manifests/init.pp]: I'd hope that we can define a regex to match available keystore types.

File: spec/acceptance/mysql_spec.rb [hunk 6]
```diff
@@ -47,25 +44,45 @@ class { 'mysql::bindings':
...
+        version         => '6.3.6',
+        downloadURL     => #{download_url},
+        javahome        => $jh,
+        db              => 'mysql',
+        dbport          => '3306',
+        dbdriver        => 'com.mysql.jdbc.Driver',
+        dbtype          => 'mysql',
+        tomcatPort      => '8081',
+        tomcatNativeSsl => true,
+        tomcatKeystoreFile => '/tmp/jira.ks',
```
Comment (ffrank) [inline spec/acceptance/mysql_spec.rb]: Some more whitespace change above?

**Comment by ffrank** [comment]
Nitpick commentary inline. Looks great otherwise :+1:

File: manifests/init.pp (hunk 4 above)
Comment (mkrakowitzer) [inline manifests/init.pp]: I Agree it needs to follow the current puppetlabs style guide, but this needs to happen as part of a major version upgrade.

I did not want to start now, and then its mixed between camelcase and underscores.

**Pull request closed without merging**